packages/drivers/odsp-driver/src/graph.ts

```diff
@@ -182,30 +215,34 @@ async function getFileDefaultUrl(
     }
 
     // ODSP link requires extra call to return link that is resistant to file being renamed or moved to different folder
-    const fetchResponse = await authorizedFetchWithRetry({
-        getToken: async (options) =>
-            getShareLinkToken(options, SharingLinkScopeFor.fileDefaultUrl, siteUrl),
-        url: `${siteUrl}/_api/web/GetFileByUrl(@a1)/ListItemAllFields/GetSharingInformation?@a1=${encodeURIComponent(
-            `'${graphItem.webDavUrl}'`,
-        )}`,
-        requestInit: {
-            method: "POST",
-            headers: {
-                "Content-Type": "application/json;odata=verbose",
-                "Accept": "application/json;odata=verbose",
+    const response = await getWithRetryForTokenRefresh(async (options) => {
+        const odspResponse = await PerformanceEvent.timedExecAsync(logger,
```

Comment:
Here and in other places: I think we usually put PerformanceEvent.timedExecAsync above getWithRetryForTokenRefresh and log number of retries it took on success. That's a bit better as you can easily count all sessions with count = 2. With logging each attempt separately, you would not be able to do same level of queries (you would be able to count number of events per session, but not number of events per call, as there maybe many calls per sessions).

Reply:
As I checked in odsp storage service, we put Performance event inside getWithRetryForTokenRefresh. So if the call throws, the performance event will be logged as canceled and on success as end. So we could still count successful calls and canceled events per session?

Reply:
I guess here we can try other way as you said and log no retries it took by logging it in count and reduce telemetry events.

Reply:
Maybe it's Ok either way. You are correct that we will get cancelation event.
The reason it's likely better to have perf event outside is that we get better handle on total time in one place, and then can rely on properties for filtering to learn more (is it because two calls?). Versus if we do it inside, we record each individually, we can get times for each call and for tocken fetch call, but have no easy way to build distribution for end-user latency.
So while it's likely more work (tracking more state and ensuring it ends on event properly), I think it's better practice.